analyzer/tools/build-logger/src/ldlogger-util.c

```diff
@@ -86,15 +86,18 @@ int predictEscapedSize(const char* str_)
 
   while (*str_)
   {
-    if (strchr("\t\b\f\r\v\n ", *str_))
-      size += 2;
+    // 0x1B is '\e' (ESC)
+    if (strchr("\a0x1B\t\b\f\r\v\n ", *str_))
+      size += 3;
```

Comment:
Can you please add an extra comment here where 3 comes from. Similarly what you did below for `size += 5`.